src/gurobi_optimods/opf/io.py

```diff
@@ -153,6 +155,7 @@ def fix_shape(arr, ncols=None):
         "gen": gen.to_dict("records"),
         "branch": branch.to_dict("records"),
         "gencost": gencost,
+        "casename": os.path.splitext(file_path)[0],
```

Comment:
Would be better to use pathlib's stem here. This also needs a test which checks for the case name.